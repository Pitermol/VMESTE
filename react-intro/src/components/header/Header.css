header {
    width: 100%;
    height: 130px;
    border: 1px;
}
#header-div {
    display: flex;
    height: 100%;
    width: 100%; 
    background-color: #FFFDC7;
    position: relative;
}
a {
    text-decoration: none;
}
.nav-block {
    display: flex;
    flex-direction: row;
    position: relative;
    bottom: 0;
    width: 65%;
    height: 50%;
    margin-top: 32.5px;
    margin-left: 4.5%;
    /* box-shadow: 0 0 20px rgba(0, 0, 0, 0.1); */
    background: #FFFDC7;
    z-index: 10;
    &--top {
        position: fixed;
        top: 0;
    }
}
.nav-elem {
    display: flex;
    justify-content: center;
    align-items: center;
    flex: 1;
    text-align: center;
    color: #000;
    letter-spacing: 0.1rem;
    transition: all 0.5s ease;
    font-size: 1.2rem;
    user-select: none;
    font-weight: 600;
    &:hover {
      color:white;
      background: rgba(220, 182, 60, 0.796);
      transition: all 0.5s ease;
    }
}
.reg-signin-block {
    display: flex;
    position: relative;
    width: 183px;
    height: 50px;
    margin-top: 40px;
    margin-left: 28px;
    border-radius: 10px;
    background-color: #E0E786;
    justify-content: center;
    align-items: center;
    font-size: 0.9rem;
    font-weight: 700;
    user-select: none;
}
.reg-signin-block .rsbtn {
    transition: 0.6s;
    &:hover {
        font-size: 1.0rem;
    }
}