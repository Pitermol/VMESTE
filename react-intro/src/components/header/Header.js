// import React from "./react";
import Logo from '../common/Logo/Logo';
import "./Header.css";
import ScrollToButton from "../common/ScrollToButton.js";
import { Link } from "react-router-dom";
import back from "../../static/header_back.png";

function NavBar() {
    return (
        <div class="nav-block">
            <ScrollToButton toId="tab-1">Особенности</ScrollToButton>
            <ScrollToButton toId="tab-2">Популярные пользователи</ScrollToButton>
            <ScrollToButton toId="tab-3">Про нас</ScrollToButton>
        </div>
    )
}

function RegSigninBtn() {
    return (
        <div class="reg-signin-block">
            <Link className='rsbtn' to="/login" style={{color: "#000"}}>Login</Link>
            <h>/</h>
            <Link className='rsbtn' to="/registration" style={{color: "#000"}}>Sign up</Link>
        </div>
    )
}

function Header(props) {
    return (
        <header>
            <div id="header-div">
                <Logo width="110"/>
                {props.isMain && <NavBar />}
                {props.isMain && <RegSigninBtn />}
            </div>
        </header>
    )
}

export default Header;
