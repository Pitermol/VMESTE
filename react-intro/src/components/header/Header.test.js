import renderer from 'react-test-renderer';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';

it('changes the class when hovered', () => {
  const component = renderer.create(
    <MemoryRouter><Header isMain={true} /></MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});