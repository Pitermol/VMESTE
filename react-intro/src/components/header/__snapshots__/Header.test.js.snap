// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<header>
  <div
    id="header-div"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height="110px"
        id="logo_img"
        src="logo.png"
        width="110px"
      />
    </a>
    <div
      class="nav-block"
    >
      <a
        className="nav-elem"
        onClick={[Function]}
      >
        Особенности
      </a>
      <a
        className="nav-elem"
        onClick={[Function]}
      >
        Популярные пользователи
      </a>
      <a
        className="nav-elem"
        onClick={[Function]}
      >
        Про нас
      </a>
    </div>
    <div
      class="reg-signin-block"
    >
      <a
        className="rsbtn"
        href="/login"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Login
      </a>
      <h>
        /
      </h>
      <a
        className="rsbtn"
        href="/registration"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Sign up
      </a>
    </div>
  </div>
</header>
`;
